app/src/main/java/com/example/androidchallenge/Activity/MainActivity.kt

```diff
@@ -166,26 +168,47 @@ private val listWeatherStatus = listOf(
         }
     }
 
-    fun getWeatherData() {
-        val weatherData = assets.open("weatherData.json")
-        var json: String? = null
-        val charset: Charset = Charsets.UTF_8
-
-        try {
-            val size = weatherData.available()
-            val buffer = ByteArray(size)
-            weatherData.read(buffer)
-            weatherData.close()
-            json = String(buffer, charset)
-        } catch (ex: IOException) {
-            ex.printStackTrace()
-        }
 
-        val gson = Gson()
-        val jsonWeather = gson.fromJson(json, WeatherJson::class.java)
-        jsonWeather?.let {  weatherViewModel.setWetherList(jsonWeather)
-            Log.d("teste", "teste ${weatherViewModel.weatherJson.value}")
-        }
+    private fun getDataWeek() {
+        val call: Call<WeatherWeekJson> = ApiWeather.getWeather
+        call.enqueue(object : Callback<WeatherWeekJson> {
+            override fun onResponse(
+                call: Call<WeatherWeekJson>?,
+                response: Response<WeatherWeekJson>?
+            ) {
+                if (response?.code() == 200) {
+                    weatherViewModel.setWetherList(response!!.body()!!)
+                }
+                else
+                    Log.d("teste", "erro ${response}")
+            }
+
+            override fun onFailure(call: Call<WeatherWeekJson>, t: Throwable) {
+                Log.d("teste", "Failure ${t}}")
+            }
+
+        })
     }
 
+    private fun getDataToday() {
+        val call: Call<Weather> = ApiWeather.getToday
+        call.enqueue(object : Callback<Weather> {
+            override fun onResponse(
+                call: Call<Weather>?,
+                response: Response<Weather>?
+            ) {
+                if (response?.code() == 200) {
+                    weatherViewModel.setTodayj(response!!.body()!!)
+                }
+                else
+                    Log.d("teste", "Erro ${response}")
+
+            }
+
+            override fun onFailure(call: Call<Weather>, t: Throwable) {
+                Log.d("teste", "Failure ${t}}")
+            }
+
+        })
+    }
```

Comment:
Since you started using basic concepts of the MVVM architecture, seems reasonable that you start giving the proper behaviour for some of your classes. Both `getDataWeek()` and `getDataToday()` are working in our "data layer" so it shouldn't be on your presentation layer. 

The ideal would be for you to move this methods to a service or repository class but, it would be enough for now if you move it to the `ViewModel`. The "shared" `ViewModel` could be a good option.